here-naksha-lib-model/src/commonMain/kotlin/naksha/model/request/PropertyFilter.kt

```
@@ -0,0 +1,89 @@
+            }
+        }
+        return false
+    }
+
+    private fun resolveEachOp(op: AnyOp, featureProperty: Any?, queryProperty: Any?) : Boolean {
+        when (op) {
+            AnyOp.EXISTS -> return featureProperty != Platform.UNDEFINED
+            AnyOp.IS_NULL -> return featureProperty == null
+            AnyOp.IS_NOT_NULL -> return featureProperty != null
+            AnyOp.IS_TRUE -> return featureProperty == true
+            AnyOp.IS_FALSE -> return featureProperty == false
+            AnyOp.IS_ANY_OF -> {
+                if (queryProperty is Array<*>) return queryProperty.contains(featureProperty)
+                if (queryProperty is List<*>) return queryProperty.contains(featureProperty)
+                return false
+            }
+            AnyOp.CONTAINS -> return resolveContains(featureProperty, queryProperty)
+            StringOp.EQUALS -> return (featureProperty is String) && (queryProperty is String) && (featureProperty.toString() == queryProperty.toString())
+            StringOp.STARTS_WITH -> return (featureProperty is String) && (queryProperty is String) && (featureProperty.startsWith(queryProperty.toString()))
+            DoubleOp.EQ -> return (featureProperty is Number) && (queryProperty is Number) && (featureProperty.toDouble() == queryProperty.toDouble())
+            DoubleOp.GT -> return (featureProperty is Number) && (queryProperty is Number) && (featureProperty.toDouble() > queryProperty.toDouble())
+            DoubleOp.LT -> return (featureProperty is Number) && (queryProperty is Number) && (featureProperty.toDouble() < queryProperty.toDouble())
+            DoubleOp.GTE -> return (featureProperty is Number) && (queryProperty is Number) && (featureProperty.toDouble() >= queryProperty.toDouble())
+            DoubleOp.LTE -> return (featureProperty is Number) && (queryProperty is Number) && (featureProperty.toDouble() <= queryProperty.toDouble())
+        }
+        return false
```

I have the same feeling about this as about `resolvePropsQuery` method - we own these types, let's be explicit about how each behaves and fail loudly if we stumble upon something we haven't expected.

Btw if we were using regular enums / sealed classes, Kotlin compiler would be able to catch this situation (unknown type) during compilation

---

As agreed per private discussion, throwing exceptions and leave a backlog item to implement the handler